static/js/chart/draw.ts

```diff
@@ -724,6 +735,80 @@ function drawLineChart(
         dots.on("click", handleDotClick);
       }
     }
+
+    if (highlightOnHover) {
+      addTooltip(id);
+      highlight.style("opacity", "0");
+      const highlightedDot = highlight
+        .append("circle")
+        .attr("r", 5)
+        .style("fill", colorFn(dataGroup.label))
+        .style("stroke", "#fff");
+      const highlightedLine = highlight
+        .append("line")
+        .attr("x1", 0)
+        .attr("y1", 0)
+        .attr("x2", 0)
+        .attr("y2", height - MARGIN.bottom)
+        .style("stroke", colorFn(dataGroup.label))
+        .style("stroke-opacity", "0.5");
+      const tooltip = d3.select(`#${id}`).select(`#${TOOLTIP_ID}`);
+      svg
+        .on("mouseover", () => {
+          highlight.style("opacity", "1");
+          tooltip.style("display", "block");
+        })
+        .on("mouseout", () => {
+          highlight.style("opacity", "0");
+          tooltip.style("display", "none");
+        })
+        .on("mousemove", () => {
+          // calculation of data point corresponding to mouse position from https://bl.ocks.org/Qizly/8f6ba236b79d9bb03a80.
```

Comment:
this could also be lifted up to a helper and is maybe a first candidate for the small refactor.

Reply:
done